### PR TITLE
OCP on AWS Tag Matching

### PR DESCRIPTION
**Pull request by adberglund**
## Summary
This should implement OCP on AWS correlation by tag matching. This updates the view for pod usage to do tag matching and adds a view for OCP on AWS storage tag matching. I also moved the OCP on AWS tables to their own folder as the OCP models file was getting long. 

## Testing
Loaded in nise data with volumes under 2 namespaces that shared a tag with AWS EBS data generated. The number of shared projects was correctly identified and cost and usage were split properly

```
GET /local/api/v1/reports/inventory/ocp/storage/?filter[time_scope_value]=-1
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "filter": {
        "time_scope_value": "-1"
    },
    "data": [
        {
            "date": "2019-02",
            "values": [
                {
                    "date": "2019-02",
                    "units": "GB-Mo",
                    "cost": 8640.0,
                    "total": 2880.0
                }
            ]
        }
    ],
    "total": {
        "units": "GB-Mo",
        "cost": 8640.0,
        "total": 2880.0
    }
}
```
```
GET /local/api/v1/reports/inventory/ocp/storage/?filter[time_scope_value]=-1&group_by[project]=*
HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "group_by": {
        "project": [
            "*"
        ]
    },
    "filter": {
        "time_scope_value": "-1"
    },
    "data": [
        {
            "date": "2019-02",
            "projects": [
                {
                    "project": "namespace_qa",
                    "values": [
                        {
                            "date": "2019-02",
                            "units": "GB-Mo",
                            "project": "namespace_qa",
                            "cost": 4320.0,
                            "total": 1440.0
                        }
                    ]
                },
                {
                    "project": "namespace_ci",
                    "values": [
                        {
                            "date": "2019-02",
                            "units": "GB-Mo",
                            "project": "namespace_ci",
                            "cost": 4320.0,
                            "total": 1440.0
                        }
                    ]
                }
            ]
        }
    ],
    "total": {
        "units": "GB-Mo",
        "cost": 8640.0,
        "total": 2880.0
    }
}
```

### CODE REVIEW

**Comment by chambridge** [comment]
Travis Error:
```
CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph: (0033_auto_20190214_1637, 0033_auto_20190214_0154 in reporting).
```
looks like the merge created a conflict with the migrations.

**Comment by adberglund** [comment]
Closed to update migrations and update distribution of number of OCP projects sharing an AWS resource in here.

File: koku/reporting/migrations/0038_auto_20190220_1511.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+                                ocp.namespace,
+                                LOWER(key) as key,
+                                LOWER(value) as value
+                            FROM reporting_ocpusagelineitem_daily as ocp,
+                                jsonb_each_text(ocp.pod_labels) labels
+                        ) AS ocp
+                            ON aws.usage_start::date = ocp.usage_start::date
+                                AND (
+                                    (aws.key = ocp.key AND aws.value = ocp.value)
+                                    OR (aws.key = 'ocp_cluster' AND aws.value = ocp.cluster_alias)
```
Comment (chambridge) [inline koku/reporting/migrations/0038_auto_20190220_1511.py]: Sorry about this, but per Dayle's comments we should use `openshift` not `ocp` so let's update these to:
- `openshift_project`
- `openshift_node`
- `openshift_cluster`

File: koku/reporting/migrations/0038_auto_20190220_1511.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+                                ocp.namespace,
+                                LOWER(key) as key,
+                                LOWER(value) as value
+                            FROM reporting_ocpstoragelineitem_daily as ocp,
+                                jsonb_each_text(ocp.persistentvolumeclaim_labels) labels
+                    ) AS pvcl
+                            ON aws.usage_start::date = pvcl.usage_start::date
+                                AND (
+                                    (aws.key = pvcl.key AND aws.value = pvcl.value)
+                                    OR (aws.key = 'ocp_cluster' AND aws.value = pvcl.cluster_alias)
```
Comment (chambridge) [inline koku/reporting/migrations/0038_auto_20190220_1511.py]: Sorry about this, but per Dayle's comments we should use `openshift` not `ocp` so let's update these to:
- `openshift_project`
- `openshift_node`
- `openshift_cluster`

File: koku/reporting/migrations/0038_auto_20190220_1511.py (hunk 1 above)
Comment (adberglund) [reply koku/reporting/migrations/0038_auto_20190220_1511.py]: Good catch!

**Pull request merged**